### ISSUE TITLE
AFE does not recognize names of the form "Func.1A"

### ISSUE THREAD

**Issue by cjhesse** [microsoft/advanced-formula-environment#13]
Prior to knowing AFE existed, I created several functions that used a similar concept of a namespace, but after the period, I used a number followed by a letter. On its own, a name can't begin with a number, but both Name Manager and LAMBDA handle my names fine.

One example is "TLU.3D"--a three-dimensional table lookup function.

It appears that AFE is recognizing "TLU" as the namespace, but when it sees "3D" it doesn't import it because it starts with a number. It does this even though "TLU.3D" meets name definition requirements.

For such named functions, AFE won't import them from name manager, nor can I import them by pasting the text. AFE doesn't give the "Incorrect name syntax" message either. Is this a feature or a bug?

**Comment by jack-williams** [comment]
Behind the scenes we model namespaces as a file, using the base name to define the name, like:
```
// in namespace TLU
3D = LAMBDA(x, ... );
```

This assumes that the base is itself a valid name. In cases where the name is not individually valid, we report an error and do not save it to the manager.

We should not "split" names into namespace / base when each part is individually not a valid name. Instead, we should treat it as atomic and put it in the top level namespace. So it would look something like:

```
// in namespace 'Workbook'
TLU.3D = LAMBDA(x, ...);
```

**Comment by cjhesse** [comment]
That sounds like a good solution to me.

I wanted to note that I created a gist with the affected functions in text form, and when I imported them that way, they all went to the workbook namespace and worked as expected. If the "hidden worksheet" option is enabled, they continue working with AFE after saving and reopening the workbook, but if the "hidden worksheet" is disabled, they won't be recognized later.
